crates/oxc_minifier/src/ast_passes/peephole_remove_dead_code.rs

```diff
@@ -124,6 +128,16 @@ impl<'a, 'b> PeepholeRemoveDeadCode {
         }
     }
 
+    /// Drop `"use strict";` directives if the function is empty.
+    fn drop_use_strict_directives_if_function_is_empty(
+        body: &mut FunctionBody<'a>,
+        _ctx: &mut TraverseCtx<'a>,
+    ) {
+        if body.statements.is_empty() {
+            body.directives.retain(|directive| !directive.is_use_strict());
+        }
+    }
```

Comment:
I'm not sure if it makes sense to exist in PeepholeRemoveDeadCode. This has to run after everything that might remove statements.